### PR TITLE
Hermann/dev

### PR DESCRIPTION
**Pull request by Hermann0222**
Boost of dds2mqtt.

### CODE REVIEW

File: nanomq_cli/dds2mqtt/dds_client.c [hunk 1]
```diff
@@ -32,17 +32,55 @@ dds_proxy(int argc, char **argv)
...
 	// mqtt_set_topics(argv[1], argv[2]);
-	mqttcli.mqttrecv_topic = "DDSCMD/HelloWorld";
-	mqttcli.mqttsend_topic = "DDS/HelloWorld";
-	ddscli.ddsrecv_topic   = "MQTTCMD/HelloWorld";
-	ddscli.ddssend_topic   = "MQTT/HelloWorld";
-
-	mqtt_connect(&mqttcli, MQTT_URL, &ddscli);
+	/* Read .conf file to get a JSON and fill topics. */
+	jso  = hocon_parse_file("/home/hermann/Documents/hermannDocuments/"
+	                         "nanomq/nanomq_cli/dds2mqtt/dds2mqtt.conf");
```
Comment (wanghaEMQ) [inline nanomq_cli/dds2mqtt/dds_client.c:50]: Hard code path?

File: nanomq_cli/dds2mqtt/dds_client.c [hunk 2]
```diff
@@ -32,17 +32,55 @@ dds_proxy(int argc, char **argv)
 {
 	mqtt_cli mqttcli;
 	dds_cli  ddscli;
+	cJSON   *jso                 = NULL;
+	cJSON   *tmp                 = NULL;
+	cJSON   *tmp2                = NULL;
+	cJSON   *tmp3                = NULL;
+	cJSON   *tmp4                = NULL;
```
Comment (wanghaEMQ) [inline nanomq_cli/dds2mqtt/dds_client.c:39]: The variable name is unreadable!

File: nanomq_cli/dds2mqtt/subscriber.c [hunk 3]
```diff
@@ -48,7 +48,7 @@ dds_subscriber(int argc, char **argv)
 
 	/* Create a Topic. */
 	topic = dds_create_topic(
-	    participant, &example_struct_desc, "MQTT/HelloWorld", NULL, NULL);
+	    participant, &example_struct_desc, argv[2], NULL, NULL);
```
Comment (wanghaEMQ) [inline nanomq_cli/dds2mqtt/subscriber.c:51]: Remember to update the helper

**Review by wanghaEMQ — CHANGES_REQUESTED** [review]

**Comment by Hermann0222** [comment]
made  varibles readable & slove the .conf file hard coding problem.

**Comment by Hermann0222** [comment]
Upate ddsproxy helper.

**Review by wanghaEMQ — APPROVED** [review]

**Review by JaylinYu — APPROVED** [review]
cool

**Pull request merged**